vendor/github.com/ONSdigital/go-ns/zebedee/client/client.go

```diff
@@ -239,3 +237,13 @@ func (c *ZebedeeClient) GetPageTitle(uri string) (data.PageTitle, error) {
 
 	return pt, nil
 }
+
+func (c *ZebedeeClient) createRequestURL(ctx context.Context, path string) string {
+	var url string
+	if ctx.Value("Collection-Id") != nil {
+		collectionID := ctx.Value("Collection-Id").(string)
```

Comment:
Same here - use the context key constant already defined and check the casting/log any error.